### PR TITLE
.github/workflows: add workflow to build Docker image

### PR DESCRIPTION
**Pull request by ueno**
This integrates a Docker image building into the GitHub actions, so the generated image can be used in our CI workflow for testing.

The workflow definition is based on:
https://docs.github.com/en/actions/publishing-packages/publishing-docker-images#publishing-images-to-github-packages

**Note that this workflow uses manual trigger (`workflow_dispatch`), which only works in the default branch on GitHub, so this PR is really intended for `main`.**

Signed-off-by: Daiki Ueno <dueno@redhat.com>

### CODE REVIEW

**Comment by derekparker** [comment]
Hey @ueno if this workflow is intended for main I don't think we'll get much out of it. The main branch really only contains a README and license file.

Is it possible to update the workflow to specify a branch to build from?

**Comment by ueno** [comment]
I guess an alternative approach would be to host the workflow in a separate repository, say `golang-fips/ci-images`.

**Review by derekparker — APPROVED** [review]
LGTM

**Comment by derekparker** [comment]
We usually require 2 approvals before merge but since this isn't specific to the actual code I'm going to go ahead and merge with only a single +1.

**Comment by derekparker** [comment]
@ueno looks like it still needs some work, when trying to run the workflow on the go1.18-openssl-fips branch I get:

```
Workflow does not exist or does not have a workflow_dispatch trigger in this branch.
```

**Comment by ueno** [comment]
> @ueno looks like it still needs some work, when trying to run the workflow on the go1.18-openssl-fips branch I get:

The idea was to use the same image for all branches. After running `workflow_dispatch` on `main`, the image will be available as `ghcr.io/golang-fips/go:main`. Then the actual testing workflow hosted in each branch could [fetch](https://github.com/golang-fips/go/pull/14/files#diff-245392b692a50c38ecab4381b118862db514035c10983f3bd4f4b7f1f4be4692R17) and use it (it seems to be [working](https://github.com/golang-fips/go/pull/14/checks) though the image is currently missing :-)

**Comment by derekparker** [comment]
> > @ueno looks like it still needs some work, when trying to run the workflow on the go1.18-openssl-fips branch I get:
> 
> The idea was to use the same image for all branches. After running `workflow_dispatch` on `main`, the image will be available as `ghcr.io/golang-fips/go:main`. Then the actual testing workflow hosted in each branch could [fetch](https://github.com/golang-fips/go/pull/14/files#diff-245392b692a50c38ecab4381b118862db514035c10983f3bd4f4b7f1f4be4692R17) and use it (it seems to be [working](https://github.com/golang-fips/go/pull/14/checks) though the image is currently missing :-)

Ah, ok I understand now, sorry for the confusion!

**Pull request merged**
